apps/src/lib/client/utils.rs

```diff
@@ -357,11 +358,12 @@ pub async fn join_network(
 const TENDERMINT_NODE_ID_LENGTH: usize = 20;
 
 /// Derive Tendermint node ID from public key
-fn id_from_pk(pk: &ed25519::PublicKey) -> TendermintNodeId {
-    let digest = Sha256::digest(pk.try_to_vec().unwrap().as_slice());
+fn id_from_pk(pk: &common::PublicKey) -> Result<TendermintNodeId, ParsePublicKeyError> {
```

Comment:
the `common::PublicKey` is not going work with Tendermint, we need to convert it to the underlying scheme before we encode it (`common::PublicKey` encoding has an extra byte prefix for the enum case)